log4j-api/src/main/java/org/apache/logging/log4j/spi/DefaultThreadContextMap.java

```diff
@@ -26,102 +27,86 @@
 import org.apache.logging.log4j.util.TriConsumer;
 
 /**
- * The actual ThreadContext Map. A new ThreadContext Map is created each time it is updated and the Map stored is always
- * immutable. This means the Map can be passed to other threads without concern that it will be updated. Since it is
- * expected that the Map will be passed to many more log events than the number of keys it contains the performance
- * should be much better than if the Map was copied for each event.
+ * The default implementation of {@link ThreadContextMap}
+ * <p>
+ *      An instance of UnmodifiableArrayBackedMap can be represented as a single {@code Object[]}, which can safely
+ *      be stored on the {@code ThreadLocal} with no fear of classloader-related memory leaks.
+ *  </p>
+ *  <p>
+ *      Performance of the underlying {@link org.apache.logging.log4j.internal.map.UnmodifiableArrayBackedMap} exceeds
+ *      {@link HashMap} in all supported operations other than {@code get()}. Note that {@code get()} performance scales
+ *      linearly with the current map size, and callers are advised to minimize this work.
+ * </p>
  */
-public class DefaultThreadContextMap implements ThreadContextMap, ReadOnlyStringMap {
-    private static final long serialVersionUID = 8218007901108944053L;
+public class DefaultThreadContextMap implements ThreadContextMap, ReadOnlyStringMap, CopyOnWrite {
```

Comment:
Can't we deprecate `CopyOnWrite` too?

Reply:
Fixed in https://github.com/apache/logging-log4j2/pull/2690/commits/50f6c36fc113bc835704ab0a9e3c594d4aaef917